auth/env_keychain_test.go

```diff
@@ -208,3 +200,76 @@ type providedAuth struct {
 func (p *providedAuth) Authorization() (*authn.AuthConfig, error) {
 	return p.config, nil
 }
+
+func useFallbackKeychain(t *testing.T) {
+	fakeKeychain := FakeKeychain{authMap: map[string]*authn.AuthConfig{
+		"no-env-auth-registry.com": {
+			Auth: "asdf=",
+		},
+	}}
+
+	resolvedKeychain, err := auth.ResolveKeychain(
+		"CNB_REGISTRY_AUTH",
+		[]string{"no-env-auth-registry.com/some-image"},
+		&fakeKeychain,
+	)
+	h.AssertNil(t, err)
+	h.AssertNil(t, os.Unsetenv("CNB_REGISTRY_AUTH")) // unset variable to prove that our test is using a pre-resolved keychain
```

Comment:
instead of moving all the test logic into these helpers could we reduce duplicated by instead, moving shared chunks like this into an `it.Before`? This seems like a `when("there is a fallback keychain", func(){}` to me and that organization would be more in keeping with the conventions in our other tests.

Reply:
We looked into this - it would mean re-ordering / re-grouping the tests such that the outer `when` is something like `when("the default keychain does not have an auth header", ...` followed by `when("valid auth env variable is set", ...` and then `when("auth env variable is not set", ...`. But to us, this ordering is less intuitive because it doesn't follow the order that the function itself is following.

Reply:
Latest changes totally solve this problem, see https://github.com/buildpacks/lifecycle/pull/409#discussion_r541151115